library/src/main/java/com/mikepenz/materialdrawer/util/Extensions.kt

```diff
@@ -0,0 +1,26 @@
+package com.mikepenz.materialdrawer.util
+
+/**
+ * Does an action if it is not null
+ */
+public inline infix fun <T> T?.ifNotNull(block: (T) -> Unit): Unit? {
+    return if (this != null) {
+        block(this)
+    } else {
+        null
+    }
+}
+
+/**
+ * Does an action if it is null
+ */
+public inline infix fun <T> T?.otherwise(block: () -> Unit): Unit = ifNull(block)
```

Comment:
Fyi, I think `Any?` will work here. You aren't using the generic in your function. Same with the method below. On the otherhand, to avoid polluting your suggestions, you can consider introducing some object, and adding the extension function on that object only. `ifNotNull` would then return that object as a nullable

Reply:
I think I will make it internal. to hide it away from others. public was to visible I think.